pandas/tests/arrays/test_datetimelike.py

```diff
@@ -324,7 +324,12 @@ def test_searchsorted_castable_strings(self, arr1d, box, string_storage):
         ):
             arr.searchsorted("foo")
 
-        arr_type = "StringArray" if string_storage == "python" else "ArrowStringArray"
+        if string_storage == "python":
+            arr_type = "StringArray"
+        elif string_storage == "pyarrow":
+            arr_type = "ArrowStringArray"
+        else:
+            arr_type = "ArrowStringArrayNumpySemantics"
```

Comment:
While this is certainly correct for this PR, it's also a quite bad error message if it includes this exact name .. We should probably replace the array type with the dtype.

Reply:
Yes, to be clear, I didn't expect at all you to change it here ;)